src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/RcTreeImporter.kt

```diff
@@ -0,0 +1,124 @@
+package org.wycliffeassociates.otter.jvm.persistence
+
+import io.reactivex.Completable
+import io.reactivex.schedulers.Schedulers
+import org.jooq.DSLContext
+import org.wycliffeassociates.otter.common.collections.tree.Tree
+import org.wycliffeassociates.otter.common.collections.tree.TreeNode
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
+import org.wycliffeassociates.otter.common.data.model.ResourceMetadata
+import org.wycliffeassociates.otter.common.domain.mapper.mapToMetadata
+import org.wycliffeassociates.otter.common.persistence.IRcTreeImporter
+import org.wycliffeassociates.otter.common.persistence.repositories.ICollectionRepository
+import org.wycliffeassociates.otter.common.persistence.repositories.IContentRepository
+import org.wycliffeassociates.otter.jvm.persistence.database.AppDatabase
+import org.wycliffeassociates.otter.jvm.persistence.entities.ResourceLinkEntity
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.CollectionMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ContentMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.LanguageMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ResourceMetadataMapper
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+
+class RcTreeImporter(
+        private val database: AppDatabase,
+        private val collectionRepository: ICollectionRepository,
+        private val contentRepository: IContentRepository
+) : IRcTreeImporter {
+
+    override fun importResourceContainer(rc: ResourceContainer, rcTree: Tree, languageSlug: String): Completable {
+        return Completable.fromAction {
+            database.transaction { dsl ->
+                val language = LanguageMapper().mapFromEntity(database.getLanguageDao().fetchBySlug(languageSlug, dsl))
+                val helpRcTargetMetadata: ResourceMetadata? =
+                        when (rc.type()) {
+                            // TODO: Identifier shouldn't just be set to "ult"
+                            "help" -> ResourceMetadataMapper().mapFromEntity(
+                                    database.getResourceMetadataDao().fetchByIdentifier("ult", dsl),
```

Comment:
@aunger To keep you in the loop, I talked to Joe about this earlier, and this is indeed what the "RC Links" table is for. Until this morning I hadn't realized that "help" resource containers can be linked to multiple "bundle" RC's (such as ULB and UDB). We decided that for now I will duplicate the content and resource entries that I am adding to the database for the notes. This is so that each entry can be linked (via the content "belongsTo" field) to the corresponding collection in either the ULB or UDB. Joe said that we are not currently using the UDB so this is a simple solution for now. Currently, I am working on the code to write entries to the "RC Links" table and to handle the general case of a variable number related resource containers.